File: packages/python-sdk/src/blindfold/regex/detectors/de/personal_id.py
"""German Personal ID (Personalausweis) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region


@register_region("de")
class PersonalIdDetector(RegexDetector):
    entity_type = EntityType.DE_PERSONAL_ID
    score = 0.75
    context_keywords = [
        "personalausweis", "personal id", "ausweis", "identifikation",
        "id-nummer", "ausweisnummer",
    ]
    context_required = True
    context_window = 50

    # New German ID format: 1 letter + 1 alphanumeric + 7 digits + 1 check digit
    pattern = re.compile(
        r"\b[CFGHJKLMNPRTVWXYZ][0-9A-Z]\d{7}\d\b"
    )

File: packages/js-sdk/src/regex/detectors/de/personal-id.ts
// German Personal ID (Personalausweis) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'

class DePersonalIdDetector extends RegexDetector {
  entityType = EntityType.DE_PERSONAL_ID
  score = 0.75
  contextKeywords = [
    'personalausweis',
    'personal id',
    'ausweis',
    'identifikation',
    'id-nummer',
    'ausweisnummer',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b[CFGHJKLMNPRTVWXYZ][0-9A-Z][0-9]{7}[0-9]\b/gi
}

registerRegion('de', DePersonalIdDetector)

export { DePersonalIdDetector }
